src/main/java/com/demo/contact/service/ContactService.java

```diff
@@ -4,119 +4,149 @@
-		return contact;
-	}
-
-	public boolean updateContact(Contact contact) {
-		String encodedImage = contact.getProfileImage();
-		contact.setProfileImage(null);
-		contact = contactRepository.save(contact);
-		if (contact != null) {
-			ContactUtils.saveImageToServer(ContactUtils.decodeToImage(encodedImage),
-					String.valueOf(contact.getContactId()));
-		}
-		return contact != null;
-	}
-
-	public Iterable<Contact> listContactsByEmail(String email) {
-		Iterable<Contact> list = contactRepository.listContactsByEmail("%" + email.trim().replace(" ", "%") + "%");
-		for (Contact contact : list) {
-			contact.add(linkTo(ContactController.class).slash(contact.getContactId()).withSelfRel());
-			contact.add(
-					linkTo(methodOn(ContactController.class).deleteContact(contact.getContactId())).withRel("delete"));
-			contact.add(linkTo(methodOn(ContactController.class).getProfilePicture(contact.getContactId()))
-					.withRel("view"));
-		}
-		return list;
-	}
-
-	public Iterable<Contact> listContactsByPhoneNumber(String phoneNumber) {
-		Iterable<Contact> list = contactRepository
-				.listContactsByPhoneNumber("%" + phoneNumber.trim().replace(" ", "%") + "%");
-		for (Contact contact : list) {
-			contact.add(linkTo(ContactController.class).slash(contact.getContactId()).withSelfRel());
-			contact.add(
-					linkTo(methodOn(ContactController.class).deleteContact(contact.getContactId())).withRel("delete"));
-			contact.add(linkTo(methodOn(ContactController.class).getProfilePicture(contact.getContactId()))
-					.withRel("view"));
-		}
-		return list;
-	}
-
-	public Iterable<Contact> listContactsByCity(String city) {
-		Iterable<Contact> list = contactRepository.listContactsByCity("%" + city.trim().replace(" ", "%") + "%");
-		for (Contact contact : list) {
-			contact.add(linkTo(ContactController.class).slash(contact.getContactId()).withSelfRel());
-			contact.add(
-					linkTo(methodOn(ContactController.class).deleteContact(contact.getContactId())).withRel("delete"));
-			contact.add(linkTo(methodOn(ContactController.class).getProfilePicture(contact.getContactId()))
-					.withRel("view"));
-		}
-		return list;
-	}
-
-	public Iterable<Contact> listContactsByState(String state) {
-		Iterable<Contact> list = contactRepository.listContactsByState("%" + state.trim().replace(" ", "%") + "%");
-		for (Contact contact : list) {
-			contact.add(linkTo(ContactController.class).slash(contact.getContactId()).withSelfRel());
-			contact.add(
-					linkTo(methodOn(ContactController.class).deleteContact(contact.getContactId())).withRel("delete"));
-			contact.add(linkTo(methodOn(ContactController.class).getProfilePicture(contact.getContactId()))
-					.withRel("view"));
-		}
-		return list;
-	}
+    @Autowired
+    private ContactRepository contactRepository;
+    @Autowired
+    private MovieScoreRepository movieScoreRepository;
+    @Autowired
+    private MovieService movieService;
```

Comment:
If you need the methods of a service in another service, you might consider merging them into one.